src/Stratis.Bitcoin/P2P/PeerSelector.cs

```diff
@@ -145,18 +153,69 @@ private IEnumerable<PeerAddress> SelectPreferredPeers()
         }
 
         /// <inheritdoc/>
-        public IEnumerable<PeerAddress> SelectPeers(int peerCount)
+        public IEnumerable<PeerAddress> SelectPeersForDiscovery(int peerCount)
         {
-            // If there are no peers, just one or if the amount peers is less than
-            // the amount of peers asked for, just return the list.
+            // If there are no peers or just one then just return the list.
             if (!this.peerAddresses.Any() || this.peerAddresses.Count == 1)
                 return this.peerAddresses.Select(pa => pa.Value);
 
-            // Randomly order the list of peers and return the amount
-            // asked for.
+            // Randomly order the list of peers and return the amount asked for.
             var allPeers = this.peerAddresses.OrderBy(p => this.random.Next());
             return allPeers.Select(p => p.Value).Take(1000);
         }
+
+        /// <inheritdoc/>
+        public IEnumerable<PeerAddress> SelectPeersForGetAddrPayload(int peerCount)
+        {
+            // If there are no peers or just one then just return the list.
+            if (!this.peerAddresses.Any() || this.peerAddresses.Count == 1)
```

Comment:
Would it be clearer to have these as two separate conditions?  If no peers, return Enumerable.Empty<PeerAddress>().  If 1 peer, return this.peerAddresses.First().Value.

Reply:
Agreed 👍